akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -304,6 +373,33 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
         settingsAppliedOk
       }
 
+      override protected def onTimer(timerKey: Any): Unit = timerKey match {
+        case Keepalive.Tick =>
+          pingState match {
+            case OptionVal.Some(state) =>
+              // don't do anything unless there are active streams
+              if (activeStreamCount() > 0) {
+                state.ticksWithoutData += 1
+                val now = System.nanoTime()
+                if (state.lastPingTimestamp > 0L && (now - state.lastPingTimestamp) > state.maxKeepaliveTimeoutNanos) {
```

Comment:
Is `lastPingTimestamp` reset somewhere (when receiving an ack)? How is it otherwise prevented that an ack is received and this logic here still triggers?

Reply:
I think I missed that reset when copying from the other PR, it should be on receiving ack.